docs/proposals/web_uploads.md

```diff
@@ -0,0 +1,34 @@
+# Add upload scheduler to webservice
+
+This is a proposal to facet `bundle upload` via the Datamon webservice (i.e. the `web` command).
+
+Additionally, so that the web service -- currently a frontend for `repo list`, `bundle list`, and `bundle list files`, all operations that typically complete in per-HTTP-request goroutines in about the expected timeframe for page loads in a 1.0-ish webservice -- can continue to be useful for a possible deploy to Kubernetes, faceting `bundle upload` necessarily includes implementing some at least rudimentary scheduling functionality:  The webservice will need to keep track of several ongoing uploads, each triggered by an HTTP-request, but with fire-and-forget (i.e. event) semantics rather than those of request-response (i.e. RPC).
+
+### data sources
+
+This proposal calls for an additional, optional flag to the `web` command, `--upload-source`, say, which can be present in arbitrary arities (to allow arbitrary numbers of upload sources -- is this possible with pflags?).  When this flag isn't present, the webservice will continue to function as before.  When a non-empty list of directories is passed in via this flag, an additional page will be available in the web ui that, at first, provides a simple file browser rooted at each of these directories.
+
+From within any navigable directory of the filebrowser-as-a-webservice, it will additionally be possible to click an "upload" button to schedule a `bundle upload` of the tree rooted at that directory.  The "upload" button will be attached to a form faceting a few additional `bundle upload` parameters:
+
+* `--concurrency-factor`
+* `--message`
+* `--repo` (dropdown)
+* `--label`
+
+### scheduler
+
+Round-robin scheduling with a heartbeat
+[as described here](https://github.com/ransomw/tubing/blob/master/src/core.clj)
+is expected to be as complicated as a first pass of the scheduler internals will be.  That is to say, every effort will be made to decomplect scheduler design at the possible expense of performance.
+
+In addition to the filebrowser views described in the data sources section, the web UI will provide a flat list of all uploads according to source directory and destination description (repo, message, label), including simple status (complete, error, ongoing).
+
+## nice-to-haves
+
+### downloadable logs
+
+The web UI will provide a flat list of all uploads, including logging  the output from `pkg/core` and beyond for download.
```

Comment:
The mention of this here makes me think I might not have understood the goal _at all_. Please clarify the intended goal or goals.

1. The functionality described here _could be_ a web UI to replace _local_ `bundle upload` actions by async uploads.
2. On the other hand, there are some hints in the text showing that _it could be_ just as well a proxy to manage a datamon instance running on a remote pod.

Reply:
indeed, it is confusing b/c the user story is about local uploads, but the web service is intended to run on a remote pod, not users' laptops.

the heart of the matter is that, real or imagined, there's sometimes a perception that "upload to service XYZ is super-easy, why can't datamon uploads be just as easy?"  with the functionality in this proposal, we can address such comments by saying "ok, continue to use the XYZ uploads your familiar with as a matter of getting the bytes from here to Iowa, then please additionally click a button over here to upload the data into datamon."

there's a feeling that, independent of technical merits, answering "can i haz?" questions in the affirmative with an "if" is usually better received than replying in the negative with a "but" as a social matter -- needless to say, i could easily be incorrect about these ui design considerations.